algorithms/duplicate.md

```diff
@@ -0,0 +1,23 @@
+---
+permalink: /algorithms/duplicate.html
+layout: default
+title: Samtools - Duplicate Marking
+highlighting: yes
+---
+## Duplicate Marking
+
+Duplicates are defined as having multiple templates whose aligned 5'
+coordinates match.  For a paired-end template this requires both 
+primary reads to have matching 5' coordinates.  Coordinates are
+based on the unclipped position of the read against the reference.
+Reads also need to match orientation.  When a duplicate is detected,
+the overall highest quality template is kept and all others have
+the duplicate flag set.
```

Comment:
@whitwham not true for Picard, as it has two ways to choose the template:
1. maximum sum of base qualities.
2. template with the most mapped bases

Picard also handles mixtures of paired end reads where either both or one end maps, preferring the template with both ends mapping, as well as mixtures of fragment (single-end) and paired end.  Then there is what happens if there is still a tie between two templates, so that we get determinism.  Picard will then compare read name and then first of pair/second of pair.

See `compare` and `fileOrderCompare` in https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/SAMRecordDuplicateComparator.java

Reply:
@nh13 do you have a link to the Picard documentation on its duplicate marking?